pagopa/io-app
# feat(Firma con IO): [SFEQS-1053] Add signature fields screen with clause selection and box drawing

## Short description
In this PR has been added a `FciSignatureFieldsScreen` where the user can see a List of clauses signature for the current document.

NOTE: to draw inside the pdf document a new library had to be introduced. To get more details about the packages used in FCI please see [this](https://pagopa.atlassian.net/wiki/spaces/SFEQS/pages/501121146/IO-REPORT+Benchmark+delle+librerie+react-native-pdf+e+pdf-lib)

https://user-images.githubusercontent.com/49342144/203784271-75ee9954-683b-4c41-8b71-ff33c2067862.mov

## List of changes proposed in this pull request
- Added a new package `pdf-lib": "^1.17.1"` [pdf-lib](https://www.npmjs.com/package/pdf-lib?activeTab=readme)
- Added `FciSignatureFieldsScreen`
- Added `SignatureFieldItem` component
- Update `FciDocumentsScreen` adding some utils to manage PDF document and drawing box for the selected signatureField
- Added an action to Abort signature request flow
- Added an action to starting signature request flow
- Added some tests

## How to test
* if not yet merged get this branch https://github.com/pagopa/io-dev-api-server/tree/SFEQS-1082-fci-endpoints from io-dev-server-api
* enable FCI_ENABLED ff
* tap on a right signature message with label WAIT_FOR_SIGNATURE
* tap on CTA of the message details
* navigate between documents using navigation bar
* scroll down the current document and tap on Continue button
* try to check some clauses
* tap on details link to show the box clause on the document
